libs/klv/src/main/java/org/codice/alliance/libs/klv/LocationKlvProcessor.java

```diff
@@ -140,12 +181,42 @@ GeoBoxHandler subsample(GeoBoxHandler geoBoxHandler, Integer subsampleCount) {
                 geoBoxHandler.getLatitude4(),
                 geoBoxHandler.getLongitude4());
 
-        geoBoxHandler.getRawGeoData()
-                .forEach((key, value) -> {
-                    for (int i = 0; i < subsampleCount; i++) {
-                        out.accept(key, value.get(i * size / subsampleCount));
-                    }
-                });
+        subsample(geoBoxHandler.getRawGeoData(), subsampleCount, size, out::accept);
+
+        return out;
+    }
+
+    private void subsample(Map<String, List<Double>> data, int subsampleCount, int size,
+            BiConsumer<String, Double> biConsumer) {
+        data.forEach((key, value) -> {
+            for (int i = 0; i < subsampleCount; i++) {
+                biConsumer.accept(key, value.get(i * size / subsampleCount));
+            }
+        });
+    }
+
+    LatitudeLongitudeHandler reduceHandlerData(LatitudeLongitudeHandler latLonHandler,
+            Integer subsampleCount) {
+
+        if (latLonHandler.getRawGeoData()
```

Comment:
code coverage in intellij shows that all the subsampling related lines are being hit